test/src/main/java/org/axonframework/test/aggregate/ResultValidatorImpl.java

```diff
@@ -431,6 +432,33 @@ private StringDescription descriptionOf(Matcher<?> matcher) {
         return this;
     }
 
+    @Override
+    public ResultValidator<T> expectExceptionDetails(Object exceptionDetails) {
+        return expectExceptionDetails(CoreMatchers.equalTo(exceptionDetails));
+    }
+
+    @Override
+    public ResultValidator<T> expectExceptionDetails(Class<?> exceptionDetails) {
+        return expectExceptionDetails(instanceOf(exceptionDetails));
+    }
+
+    @Override
+    public ResultValidator<T> expectExceptionDetails(Matcher<?> exceptionDetailsMatcher) {
+        StringDescription emptyMatcherDescription = new StringDescription(
+                new StringBuilder("Given exception details matcher is null!"));
+        if (exceptionDetailsMatcher == null) {
+            reporter.reportWrongExceptionDetails(actualException, emptyMatcherDescription);
+            return this;
+        }
+        StringDescription description = new StringDescription();
+        exceptionDetailsMatcher.describeTo(description);
+        if (actualException != null && !exceptionDetailsMatcher.matches(HandlerExecutionException.resolveDetails(
```

Comment:
If the `actualException` is `null`, I would expect this validation to fail, as by nature there wouldn't be any details to expect in that case.
Hence, I think an additional if-block, as with the `expectException(Matcher)` method, would be in place.

I spot the `expectExceptionMessage` isn't doing this either, but that's a bug I'm afraid.

Reply:
See my other [comment](https://github.com/AxonFramework/AxonFramework/pull/2125#discussion_r820523593)